cmd/util/ledger/migrations/cadence_values_migration.go

```diff
@@ -167,13 +174,29 @@ func (m *CadenceBaseMigrator) MigrateAccount(
 		flow.ConvertAddress(address): {},
 	}
 
-	return MergeRegisterChanges(
+	newPayloads, err := MergeRegisterChanges(
 		migrationRuntime.Snapshot.Payloads,
 		result.WriteSet,
 		expectedAddresses,
 		expectedAddresses,
 		m.log,
 	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to merge register changes: %w", err)
+	}
+
+	if m.diffReporter != nil {
+
+		accountDiffReporter := NewCadenceValueDiffReporter(address, m.diffReporter, m.logVerboseDiff)
+
+		err = accountDiffReporter.DiffStates(oldPayloads, newPayloads, domains)
```

Comment:
Since this is a diagnostics tool, it might make sense for `accountDiffReporter. DiffStates` not to return an error, and just log it internally.

Similar for the `m.diffReporter != nil` check.

Reply:
Good point!  :+1: Done in commit 80e915c60a18ee685b60b18a93f3c6b648abfebb.